### PR TITLE
Use integration commit for calculating current diffs

### PR DESCRIPTION
**Pull request by mtsgrd**
I investigated why you sometimes see pairs of inverse hunks when working on more than one branch. The main takeaway is that we've been needlessly including diffs for already committed code in the calculations about what uncommitted hunks belong to what branch.

When use the right base then we don't need any complicated `calculate_uncommitted_files`, we simply assign uncommitted hunks based on either branch lock or ownership claim.

### CODE REVIEW

File: crates/gitbutler-core/src/virtual_branches/virtual.rs [hunk 1]
```diff
@@ -833,19 +838,18 @@ pub fn list_virtual_branches(
...
         .iter()
         .filter_map(|(branch, _)| branch.selected_for_changes)
         .max()
         .unwrap_or(-1);
 
     for (branch, files) in &statuses {
-        // check if head tree does not match target tree
-        // if so, we diff the head tree and the new write_tree output to see what is new and filter the hunks to just those
-        let files =
-            calculate_non_commited_diffs(project_repository, branch, &default_target, files)?;
```
Comment (krlvi) [inline crates/gitbutler-core/src/virtual_branches/virtual.rs:847]: Trying to test this out and understand, does skipping this work? Or does `get_status_by_branch` give us something else now?

Comment (krlvi) [reply crates/gitbutler-core/src/virtual_branches/virtual.rs:847]: Just saw the commit message, what do you mean by produces it's own input? (there is shadowing of the local var)

Comment (mtsgrd) [reply crates/gitbutler-core/src/virtual_branches/virtual.rs:847]: By producing its own input, I mean it did something complicated that returned its original input. That and it resolved conflict markers, I missed that, but also just fixed it.

*edit: it makes sense if you think of it from the perspective that now we start out with only uncommitted hunks, so no need to calculate them.

File: crates/gitbutler-core/tests/suite/virtual_branches/unapply.rs [hunk 2]
```diff
@@ -142,7 +142,7 @@ async fn conflicting() {
         assert!(!branch.active);
         assert!(!branch.base_current);
         assert!(!branch.conflicted);
-        assert_eq!(branch.files[0].hunks[0].diff, "@@ -1 +1 @@\n-first\n\\ No newline at end of file\n+conflict\n\\ No newline at end of file\n");
+        assert_eq!(branch.files[0].hunks[0].diff, "@@ -1 +1 @@\n-second\n\\ No newline at end of file\n+conflict\n\\ No newline at end of file\n");
```
Comment (mtsgrd) [inline crates/gitbutler-core/tests/suite/virtual_branches/unapply.rs]: This one doesn't feel right, need to debug still. *edit: actually, I think it does.. when unapplied, [we are diffing](https://github.com/gitbutlerapp/gitbutler/blob/master/crates/gitbutler-core/src/virtual_branches/virtual.rs#L2104) against `default_target` which has now updated to include commit 2.

File: crates/gitbutler-core/tests/suite/virtual_branches/apply_virtual_branch.rs [hunk 3]
```diff
@@ -124,7 +124,7 @@ async fn rebase_commit() {
         let (branches, _, _) = controller.list_virtual_branches(project_id).await.unwrap();
         assert_eq!(branches.len(), 1);
         assert_eq!(branches[0].id, branch1_id);
-        assert_eq!(branches[0].files.len(), 0);
+        assert_eq!(branches[0].files.len(), 1);
```
Comment (mtsgrd) [inline crates/gitbutler-core/tests/suite/virtual_branches/apply_virtual_branch.rs]: Looks wrong, I'm debugging these.

**Pull request merged**